### PR TITLE
Fix env variables and change ports

### PR DESCRIPTION
**Pull request by mmiszy**
Fixes https://github.com/saleor/saleor-checkout/issues/171

# This is a breaking change for development

This PR changes ports where our apps are running. This is necessary to align with the (soon to be merged) Saleor React Storefront repo.

- `apps/react-storefront`: 3000 (soon to be merged)
- `apps/saleor-app-checkout`: 3001
- `packages/checkout-storefront`: 3002
- `apps/checkout`: 3003 (deprecated)
- `packages/ui-kit storybook`: 6006

This PR also fixes the problem where root-level env variables from `.env.local` file were not loaded in `dev`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -163,3 +163,11 @@ Read setup guide in [docs/adyen.md](./docs/adyen.md)
...
 
 Read FAQ in [docs/faq.md](./docs/faq.md)
+
+## Ports
+
+- apps/react-storefront: 3000
+- apps/saleor-app-checkout: 3001
+- packages/checkout-storefront: 3002
+- apps/checkout: 3003
+- packages/ui-kit storybook: 6006
```
Comment (witoszekdev) [inline README.md:173]: Smol formatting suggestion ;)
```suggestion
- `apps/react-storefront`: **3000**
- `apps/saleor-app-checkout`: **3001**
- `packages/checkout-storefront`: **3002**
- `apps/checkout`: **3003**
- `packages/ui-kit storybook`: **6006**
```

File: apps/saleor-app-checkout/constants.ts [hunk 2]
```diff
@@ -14,12 +14,12 @@ export type DebugEnvVars = Record<DebugEnvVar, string | undefined>;
 
 // Need to use `var variable = process.env.VARIABLE;`, not `var env = process.env; var variable = env.VARIABLE;`
 // https://github.com/vercel/next.js/issues/19420
+export const envVars: EnvVars = {
+  apiUrl: process.env["NEXT_PUBLIC_SALEOR_API_URL"]!,
```
Comment (witoszekdev) [inline apps/saleor-app-checkout/constants.ts:18]: We can't use process.env like that, due to how Next.js env variables processing works - I'm working on fix here: https://github.com/saleor/saleor-checkout/pull/174

```suggestion
  apiUrl: process.env.NEXT_PUBLIC_SALEOR_API_URL!,
```

File: apps/saleor-app-checkout/package.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
   "version": "0.0.0",
   "private": true,
   "scripts": {
-    "dev": "NODE_OPTIONS='--inspect' next dev",
+    "dev": "NODE_OPTIONS='--inspect' next dev -p 3001",
```
Comment (witoszekdev) [inline apps/saleor-app-checkout/package.json:6]: We should also add the same port for `build` script

File: package.json [hunk 4]
```diff
@@ -9,7 +9,7 @@
   "scripts": {
     "build": "turbo run build",
     "generate": "turbo run generate",
-    "dev": "env-cmd turbo run dev --parallel",
+    "dev": "env-cmd -f .env env-cmd -f .env.local turbo run dev --parallel",
```
Comment (witoszekdev) [inline package.json:12]: Do I understand this correctly?
- it loads `.env` and `.env.local` files from the root of monorepo
- each package should also load their env vars by using `packages/env-vars` in the `dev` script (for example NODE_OPTIONS=`--require env-vars`)?

File: apps/saleor-app-checkout/constants.ts (hunk 2 above)
Comment (mmiszy) [reply apps/saleor-app-checkout/constants.ts:18]: Yes, we can. We can't use variables to get anything front-end related from `process.env` but `process.env["NEXT_PUBLIC_SALEOR_API_URL"]` is fine

File: package.json (hunk 4 above)
Comment (mmiszy) [reply package.json:12]: It loads the variables from the root.
Other packages should use `dotenv` or `env-cmd` or whatever to load their local variables.

File: apps/saleor-app-checkout/package.json (hunk 3 above)
Comment (mmiszy) [reply apps/saleor-app-checkout/package.json:6]: Done

File: package.json (hunk 4 above)
Comment (witoszekdev) [reply package.json:12]: One more thing - it doesn't work if I don't use have `.env.local` file. By default it's not stored in the repository, so the script ideally should ignore that file if it doesn't exist, so it just works™️ after cloning the repo

File: apps/saleor-app-checkout/constants.ts (hunk 2 above)
Comment (witoszekdev) [reply apps/saleor-app-checkout/constants.ts:18]: but... it is frontend realted, this env variable it's used for creating urql client on frontend

File: package.json (hunk 4 above)
Comment (mmiszy) [reply package.json:12]: Good catch @witoszekdev

Comment (mmiszy) [reply package.json]: Fixed

File: apps/saleor-app-checkout/constants.ts (hunk 2 above)
Comment (mmiszy) [reply apps/saleor-app-checkout/constants.ts:18]: We talked in direct messages, it's all good!

**Review by witoszekdev — APPROVED** [review]

**Pull request merged**